### PR TITLE
Added gen2std miniapp (no check)

### PR DESCRIPTION
**Pull request by rasolca**
Added miniapp for genToStd.

Currently no check is implemented, as it would require triangular matrix-matrix multiplication.

A temporary check could be implemented in the following way:
genToStd: A,B -> C = B^-1 A B^-H
trsm:         B, Id -> D = B^-1 Id = B^-1
genToStd  C, D -> E = D^-1 C D^-H = B B^-1 A B^-H B^H = A

### CODE REVIEW

File: miniapp/miniapp_gen_to_std.cpp [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+      dlaf::common::Timer<> timeit;
+      dlaf::eigensolver::genToStd<Backend::Default, Device::Default, T>(comm_grid, blas::Uplo::Lower,
+                                                                        matrix_a.get(), matrix_b.get());
+
+      // wait and barrier for all ranks
+      {
+        matrix_a.get().waitLocalTiles();
+        DLAF_MPI_CALL(MPI_Barrier(world));
+      }
```
Comment (albestro) [inline miniapp/miniapp_gen_to_std.cpp:132]: does this scope serve any purpose?

**Review by albestro — APPROVED** [review]

File: miniapp/miniapp_gen_to_std.cpp [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      std::cout << "[" << run_index << "]"
+                << " " << elapsed_time << "s"
+                << " " << gigaflops << "GFlop/s"
+                << " " << matrix_a_host.size() << " " << matrix_a_host.blockSize() << " "
+                << comm_grid.size() << " " << hpx::get_os_thread_count() << std::endl;
+
+    // (optional) run test
+    if ((opts.do_check == CheckIterFreq::Last && run_index == (opts.nruns - 1)) ||
+        opts.do_check == CheckIterFreq::All) {
+      // TODO
```
Comment (albestro) [inline miniapp/miniapp_gen_to_std.cpp:152]: `DLAF_UNIMPLEMENTED`?

**Review by albestro — APPROVED** [review]

**Review by albestro — APPROVED** [review]
I just quickly reread the code, nothing come up to my eye. LGTM

**Comment by rasolca** [comment]
bors try

**Comment by albestro** [comment]
bors try

**Comment by albestro** [comment]
bors try

**Comment by rasolca** [comment]
bors try

**Pull request merged**